### ISSUE TITLE
Tuple2 pickling

### ISSUE THREAD

**Issue by mikkka** [scala/pickling#348]
pickle version 0.10.1

> import scala.pickling._
> import scala.pickling.Defaults._
> import scala.pickling.json._
> import scala.pickling.static._
> 
> (42,42).pickle

scala.pickling.PicklingException: Class class scala.Tuple2$mcII$sp not recognized by pickler, looking for one of: scala.Tuple2[scala.Int,scala.Int]
  at ScalaTuple2$u005BscalaInt$u002CscalaInt$u005DPickler$macro$22$2$.pickle(<console>:29)
  at ScalaTuple2$u005BscalaInt$u002CscalaInt$u005DPickler$macro$22$2$.pickle(<console>:29)
  at scala.pickling.functions$.pickleInto(functions.scala:36)
  at scala.pickling.functions$.pickle(functions.scala:19)
  at scala.pickling.PickleOps.pickle(Ops.scala:9)
  ... 43 elided

**Comment by mikkka** [comment]
> import scala.pickling._
> import scala.pickling.Defaults._
> import scala.pickling.binary._
> 
> (42,42).pickle.value.unpickle[(Int,Int)]

res1: (Int, Int) = (9,1935892844)

**Comment by jsuereth** [comment]
Looks to be an issue with specialization.  Thanks for the report!

**Comment by mikkka** [comment]
Yes, it's problem with specialization

**Comment by jsuereth** [comment]
Fixed on 0.11.x branch (with new macros):

```
scala> import scala.pickling._
import scala.pickling._

scala> import scala.pickling.Defaults._
import scala.pickling.Defaults._

scala> import scala.pickling.binary._
import scala.pickling.binary._

scala> (42,42).pickle
res0: pickling.binary.pickleFormat.PickleType = BinaryPickle([0,0,0,33,115,99,97,108,97,46,84,117,112,108,101,50,91,115,99,97,108,97,46,65,110,121,44,115,99,97,108,97,46,65,110,121,93,0,0,0,9,115,99,97,108,97,46,73,110,116,0,0,0,42,0,0,0,9,115,99,97,108,97,46,73,110,116,0,0,0,42])

scala> (42,42).pickle.unpickle[(Int, Int)]
res1: (Int, Int) = (42,42)
```
